x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -150,7 +150,10 @@ export function App({
     }
   }, [docId]);
 
-  const isSaveable = lastKnownDoc && core.application.capabilities.visualize.save;
+  const isSaveable =
+    lastKnownDoc &&
+    lastKnownDoc.expression.length > 0 &&
```

Comment:
This test looks like it could throw if `expression` is null instead of `""`

Reply:
According to the type definition, it will never be null, only string:
https://github.com/elastic/kibana/blob/9769be3a289b4783a59988e9abf0cc2a5a1b6ba4/x-pack/legacy/plugins/lens/public/persistence/saved_object_store.ts#L16

Thoughts?